Description: The filename of the library f2fs_dlutils.so will dlopen is
             changed.
Author: Kai-Chung Yan <seamlikok@gmail.com>
Last-Update: 2015-07-21
--- a/extra/f2fs_utils/f2fs_dlutils.c
+++ b/extra/f2fs_utils/f2fs_dlutils.c
@@ -35,7 +35,8 @@
 
 #include <f2fs_fs.h>
 #include <f2fs_format_utils.h>
-#define F2FS_DYN_LIB "libf2fs_fmt_host_dyn.so"
+#define F2FS_DYN_LIB "/usr/lib/android/libf2fs.so"
+#define F2FS_FORMAT_DYN_LIB "/usr/lib/android/libf2fs_format.so"
 
 int (*f2fs_format_device_dl)(void);
 void (*f2fs_init_configuration_dl)(struct f2fs_configuration *);
@@ -51,12 +52,14 @@
 
 int dlopenf2fs() {
 	void* f2fs_lib;
+	void* f2fs_format_lib;
 
 	f2fs_lib = dlopen(F2FS_DYN_LIB, RTLD_NOW);
-	if (!f2fs_lib) {
+	f2fs_format_lib = dlopen(F2FS_FORMAT_DYN_LIB, RTLD_NOW);
+	if ((!f2fs_lib) || (!f2fs_format_lib)) {
 		return -1;
 	}
-	f2fs_format_device_dl = dlsym(f2fs_lib, "f2fs_format_device");
+	f2fs_format_device_dl = dlsym(f2fs_format_lib, "f2fs_format_device");
 	f2fs_init_configuration_dl = dlsym(f2fs_lib, "f2fs_init_configuration");
 	if (!f2fs_format_device_dl || !f2fs_init_configuration_dl) {
 		return -1;
